File: leetcode/solutions/1009.ComplementOfBase10Integer/bitwiseComplement_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import bitwiseComplement from "./bitwiseComplement.ts";

Deno.test("1009. Complement of Base 10 Integer", () => {
  type test = [number, number];
  const tests: test[] = [
    [0, 1],
    [5, 2],
    [7, 0],
    [10, 5],
  ];

  for (const [arg, want] of tests) {
    assertEquals(bitwiseComplement(arg), want);
  }
});

File: leetcode/solutions/1009.ComplementOfBase10Integer/bitwiseComplement_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_1009_ComplementOfBase10Integer(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  int
		out int
	}{
		{0, 1},
		{5, 2},
		{7, 0},
		{10, 5},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := bitwiseComplement(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
